felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -1057,6 +1099,43 @@ func (m *bpfEndpointManager) onInterfaceUpdate(update *ifaceStateUpdate) {
 		return
 	}
 
+	if !m.isWorkloadIface(update.Name) {
+		// Determine the type of interface.
+		// These include host, bond, slave, ipip, wireguard, l3.
+		// update the nameToIfaceTypeMap
+		curIfaceType, masterIndex, err := m.dp.getIfaceType(update.Name)
+		if err != nil || curIfaceType == IfaceTypeUnknown {
+			log.Panicf("Failed to get interface information via netlink '%s'", update.Name)
+		} else {
+			prevIfaceType := IfaceTypeUnknown
+			if val, ok := m.nameToIfaceType[update.Name]; ok {
+				prevIfaceType = val
+			}
+			if prevIfaceType != curIfaceType {
+				if curIfaceType == IfaceTypeBondSlave {
+					// Remove the Tc program.
+					ai, err := bpf.ListCalicoAttached()
+					if err != nil {
+						log.WithError(err).Warn("Failed to list attached programs")
+					} else {
+						if err := m.cleanupOldAttach(update.Name, ai[update.Name]); err != nil {
+							log.WithError(err).Warnf("Failed to detach old programs from now bonding device '%s'", update.Name)
+						}
+					}
+					// Check if the master device matches the regex.
+					masterIfa, err := net.InterfaceByIndex(masterIndex)
+					if err != nil {
+						log.Debugf("Failed to get master interface details for '%s'", update.Name)
+					}
+					if !m.isDataIface(masterIfa.Name) {
+						log.Warnf("Master interface '%s' ignored. Add it to the config", masterIfa.Name)
```

Comment:
I think the better behaviour would be to say "ok, master is not in the regexp but the slave is, seems like the user wants us to attach to the slave and not to the bond. User may know better, why not, lets to it. Attach to slave and ignore the master" WDYT? :thinking: